mitmproxy/addons/save.py

```diff
@@ -88,28 +89,24 @@ def tcp_end(self, flow):
     def tcp_error(self, flow):
         self.tcp_end(flow)
 
-    def websocket_start(self, flow):
-        if self.stream:
-            self.active_flows.add(flow)
-
-    def websocket_end(self, flow):
+    def websocket_end(self, flow: http.HTTPFlow):
         if self.stream:
             self.stream.add(flow)
             self.active_flows.discard(flow)
 
-    def websocket_error(self, flow):
+    def websocket_error(self, flow: http.HTTPFlow):
         self.websocket_end(flow)
 
-    def request(self, flow):
+    def request(self, flow: http.HTTPFlow):
         if self.stream:
             self.active_flows.add(flow)
 
-    def response(self, flow):
-        if self.stream:
+    def response(self, flow: http.HTTPFlow):
+        if self.stream and flow.websocket is None:
```

Comment:
Does this mean the HTTP 101 flow is not saved and lost? But it contains interesting data such as authentication tokens. Or why was that change needed?

Reply:
For a moment I must've forgotten that `WebSocketFlow` is gone, makes sense now